src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,36 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (obj.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle = (Rectangle) obj;
+            return color == rectangle.color || color != null && color.equals(rectangle.color)
+                    && length == rectangle.length
+                    || length != null && length.equals(rectangle.length)
+                    && width == rectangle.width
+                    || width != null && width.equals(rectangle.width);
```

Comment:
The logical-AND operator ( && ) has higher precedence than the logical-OR operator ( || )
add parenthesis to this statement to ensure the flow is correct